website/static/js/remote/knockout.remote.js

```diff
@@ -42,30 +42,139 @@ function onWsError() {
   loginVM.wsError(true);
 }
 
+
+function updatePlaylistTrack(newPlaylistTracks) {
+  mappedPlaylistTracks = $.map(newPlaylistTracks, function(item) {
+    return new PlaylistTrack(item);
+  });
+  roomVM.playlistTracks(mappedPlaylistTracks);
+}
+
 // receive a message though the websocket from the server
 function receiveMessage(message) {
-  if(message.stop) {
-    // Stop all players
-    Object.keys(playerControlWrapper).forEach(function(player) {
-      playerControlWrapper[player].stop();
-      $('.player-child').not('.player-child-no-music').stop().fadeOut(250);
-    });
-  }
-  if(message.action) {
-    if(message.action !== "play" || message.action === "play" && roomVM.playerOpen()) {
-      // stop all others players
+  switch(message.action) {
```

Comment:
May split each case in a function, then call them in the switch. It avoid a 180 lines function to maintain